extensions/ql-vscode/src/remote-queries/view/RemoteQueries.tsx

```diff
@@ -111,13 +127,11 @@ export function RemoteQueries(): JSX.Element {
         </span>
       </p>
 
-      <div className="vscode-codeql__query-summary-container">
-        <h2 className="vscode-codeql__query-summary-title">Repositories with results ({queryResult.affectedRepositoryCount}):</h2>
-        <a className="vscode-codeql__summary-download-link vscode-codeql__download-link"
-          onClick={() => download(queryResult.downloadLink)}>
-          {octicons.download}Download all
-        </a>
-      </div>
+      {
+        queryResult.affectedRepositoryCount === 0
+          ? <SummaryNoResults />
+          : <SummaryWithResults {...queryResult} />
```

Comment:
Is there a reason to create a copy of `queryResult` here? I guess it's safer than being able to accidentally tamper with the object. Just want to know what you're thoughts are here.

Reply:
I could be wrong, but I think the alternative way of doing this would mean having to write all the property names out. Using the spread operator creates all the props based on the object we pass. So it's about readability rather than immutability.